src/common/utils/toggle-set.ts

```diff
@@ -0,0 +1,24 @@
+import { action, ObservableSet } from "mobx";
+
+export class ToggleSet<T> extends Set<T> {
+  public toggle(value: T): void {
+    if (!this.delete(value)) {
+      // Set.prototype.delete returns false if `value` was not in the set
+      this.add(value);
+    }
+  }
+}
+
+export class ObservableToggleSet<T> extends ObservableSet<T> {
+  @action
+  public toggle(value: T): void {
+    if (!this.delete(value)) {
+      // Set.prototype.delete returns false if `value` was not in the set
+      this.add(value);
+    }
+  }
+
+  toJS(): ToggleSet<T> {
```

Comment:
`mobx6` returns an array of values instead of `Set`. Maybe better to return `mobx.toJS(this)`?

Reply:
Yes good point. Thanks